pages/dashboard/DisplayInvitations.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useRouter } from 'next/router';
+
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import TextButton from '../../components/UI/TextButton';
+import AcceptIcon from '../../assets/icons/accept.svg';
+import CloseIcon from '../../assets/icons/close.svg';
+import LoadingSpinnerInline from '../../components/UI/LoadingSpinnerInline';
+
+/**
+ * COMPONENT 'DisplayInvitations'
+ * Displays an invitation for a matrix room/space within a <ServiceTable> component and gives users the option to accept or decline them.
+ *
+ * @param {Object} invite — object of the room the user was invited to
+ * @param {String} service — name of the service (i.e. the object key names of the 'authProvider' object in the config)
+ * @param {String} path — name of the Application (i.e. the 'path' variable in the config)
+ *
+ * @callback acceptMatrixInvite - callback function to accept invitations to a matrix room or space
+ * @param roomId - matrix roomId
+ * @param service - name of the service (parsed to the function)
+ *
+ * @callback declineMatrixInvite - callback function to decline invitations to a matrix room or space
+ * @param roomId - matrix roomId
+ *
+ * @returns {React.ReactComponent}
+*/
+
+const DisplayInvitations = ({ invite, path, service, acceptMatrixInvite, declineMatrixInvite }) => {
+    const { t } = useTranslation('dashboard');
+    const [isAcceptingInvite, setIsAcceptingInvite] = useState(false);
+    const [isDecliningInvite, setIsDecliningInvite] = useState(false);
+    const router = useRouter();
+
+    const handleDecline = async (e, roomId) => {
+        e.preventDefault();
+        setIsDecliningInvite(true);
+        await declineMatrixInvite(roomId);
+        setIsDecliningInvite(false);
+    };
+
+    const handleAccept = async (e, roomId) => {
+        e.preventDefault();
+        setIsAcceptingInvite(true);
+        const forwardingUrl = await acceptMatrixInvite(roomId, service);
+        setIsAcceptingInvite(false);
+        if (forwardingUrl) {
+            if (confirm('You’ve successfully accepted the invitation!\n\nWould you like to be redirected to the newly accepted {{name}} item?', { name: path })) {
```

Comment:
I'm not sure if I like this... let's assume I have 10 invitations. So instead of being able to quickly accept/reject them, I have to confirm or deny this after every step.

It's more like this: either I want to be forwarded to the item I just accepted the invitation for, or I don't and I want to stay on my dashboard. But if I choose the latter, why would I decide otherwise after accepting/rejecting the next invitation?

I'm not sure what to suggest but removing this `confirm()` altogether.
Or somehow memorizing if the user said "no, don't forward me" and then not doing so for all subsequent accepts.

What do you think from an UX POV @andirueckel?

Reply:
Also, something's not working:
<img width="413" alt="image" src="https://github.com/medienhaus/medienhaus-spaces/assets/706419/57a37d2f-9a43-47c7-a5e2-4f50119280ff">

Reply:
> It's more like this: either I want to be forwarded to the item I just accepted the invitation for, or I don't and I want to stay on my dashboard. But if I choose the latter, why would I decide otherwise after accepting/rejecting the next invitation?

@fnwbr @marcel-klasse Let’s replace the `confirm()` with `alert()` and remove the ”Would you like to be redirected …” paragraph. After having accepted the invitation for/to an item, let’s disable the respective accept/decline buttons and make the item’s display name a proper hyperlink.